modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java

```diff
@@ -0,0 +1,338 @@
+
+    /**
+     * @return SMTP properties of Gmail server
+     */
+    public static Properties getSMTPProperties() {
+        Properties props = new Properties();
+        props.put("mail.smtp.host", "smtp.gmail.com");
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.debug", "true");
+        props.put("mail.smtp.socketFactory.port", "465");
+        props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+        props.put("mail.smtp.socketFactory.fallback", "false");
+        props.put("mail.smtp.starttls.enable", "true");
+        return props;
+    }
+
+    /**
+     * Send email and check email received successfully
+     *
+     * @param emailSubject - Massage to send in email
+     * @return Email - Email sent successfully or not
+     * @throws Exception - Error while sending the gmail
+     */
+    public static boolean sendMailAndCheckReceived(String emailSubject) throws Exception {
+        Properties props = getSMTPProperties();
+        boolean isEmailReceived = false;
+        EmailSender emailSender =
+                new EmailSender(props, sender, String.valueOf(senderPassword), domain, receiver + "@" + domain);
+        if (emailSender.createSession()) {
+            emailSender.setSubject(emailSubject);
+            emailSender.setBody("Body : " + emailSubject);
+            emailSender.sendEmail();
+            log.info("Email send by Mail API successfully : " + emailSubject);
+            isEmailReceived = waitToCheckEmailReceived(emailSubject,EMAIL_INBOX);
+        }
+        return isEmailReceived;
+    }
+
+    /**
+     * Check a particular email has received to a given email folder by email subject.
+     *
+     * @param emailSubject - Email emailSubject to find email is in inbox or not
+     * @return - found the email or not
+     * @throws ESBMailTransportIntegrationTestException - Is thrown if an error occurred while reading the emails
+     */
+    public static boolean isMailReceivedBySubject(String emailSubject, String folder)
+            throws ESBMailTransportIntegrationTestException {
+        Store store = null;
+        boolean emailReceived = false;
+        Folder mailFolder;
+        try {
+            store = getConnection();
+            mailFolder = store.getFolder(folder);
+            mailFolder.open(Folder.READ_WRITE);
+            SearchTerm searchTerm = new AndTerm(new SubjectTerm(emailSubject), new BodyTerm(emailSubject));
+            Message[] messages = mailFolder.search(searchTerm);
+            for (Message message : messages) {
+                if (message.getSubject().contains(emailSubject)) {
+                    log.info("Found the email emailSubject : " + emailSubject);
+                    emailReceived = true;
+                    break;
+                }
+            }
+            return emailReceived;
+        } catch (MessagingException ex) {
+            log.error("Error when getting mail count ", ex);
```

Comment:
I was logging the getConnection() method's exception in two other places . Now i have fixed it,